contracts/poap-manager/src/contract.rs

```diff
@@ -0,0 +1,530 @@
+    let admin = deps.api.addr_validate(&msg.admin)?;
+    let config = Config {
+        admin: admin.clone(),
+        poap_code_id: msg.poap_code_id.u64(),
+    };
+    CONFIG.save(deps.storage, &config)?;
+
+    // assign the minter of poap contract to the manager's contract address
+    let mut poap_instantiate_msg = msg.poap_instantiate_msg;
+    poap_instantiate_msg.minter = env.contract.address.into();
+
+    Ok(Response::new()
+        .add_attribute("action", ACTION_INSTANTIATE)
+        .add_attribute(ATTRIBUTE_ADMIN, admin)
+        .add_attribute(ATTRIBUTE_POAP_CODE_ID, msg.poap_code_id)
+        .add_submessage(SubMsg::reply_on_success(
+            wasm_instantiate(
+                msg.poap_code_id.u64(),
+                &poap_instantiate_msg,
+                info.funds,
+                "poap".into(),
+            )?,
+            INSTANTIATE_POAP_REPLY_ID,
+        )))
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn reply(
+    deps: DepsMut<DesmosQuery>,
+    _env: Env,
+    msg: Reply,
+) -> Result<Response<DesmosMsg>, ContractError> {
+    match msg.id {
+        INSTANTIATE_POAP_REPLY_ID => resolve_instantiate_poap_reply(deps, msg),
+        _ => Err(ContractError::InvalidReplyID {}),
+    }
+}
+
+fn resolve_instantiate_poap_reply(
+    deps: DepsMut<DesmosQuery>,
+    msg: Reply,
+) -> Result<Response<DesmosMsg>, ContractError> {
+    let res = parse_reply_instantiate_data(msg)?;
+    let address = deps.api.addr_validate(&res.contract_address)?;
+    POAP_CONTRACT_ADDRESS.save(deps.storage, &address)?;
+    Ok(Response::new().add_attribute(ATTRIBUTE_ACTION, ACTION_INSTANTIATE_POAP_REPLY))
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    deps: DepsMut<DesmosQuery>,
+    _env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response<DesmosMsg>, ContractError> {
+    match msg {
+        ExecuteMsg::Claim {} => execute_claim(deps, info),
+        ExecuteMsg::MintTo { recipient } => execute_mint_to(deps, info, recipient),
+        ExecuteMsg::UpdateAdmin { new_admin } => execute_update_admin(deps, info, new_admin),
+    }
+}
+
+fn execute_claim(
+    deps: DepsMut<DesmosQuery>,
+    info: MessageInfo,
+) -> Result<Response<DesmosMsg>, ContractError> {
+    let poap_contract_address = POAP_CONTRACT_ADDRESS.load(deps.storage)?;
+    if !check_eligibility(deps, info.sender.clone())? {
+        return Err(ContractError::NoEligibilityError {});
+    }
+    Ok(Response::new()
+        .add_attribute(ATTRIBUTE_ACTION, ACTION_CLAIM)
+        .add_attribute(ATTRIBUTE_SENDER, &info.sender)
+        .add_message(wasm_execute(
+            poap_contract_address,
+            &POAPExecuteMsg::MintTo {
+                recipient: info.sender.into(),
+            },
+            info.funds,
+        )?))
+}
+
+fn check_eligibility(deps: DepsMut<DesmosQuery>, user: Addr) -> Result<bool, ContractError> {
+    ProfilesQuerier::new(deps.querier.deref()).query_profile(user)?;
```

Comment:
I think we can have a function returning a simple bool.
Can be something like this:
``` rust
ProfilesQuerier::new(deps.querier.deref()).query_profile(user).is_ok()
```
Otherwise when `query_profile` fails because the user don't have a profile the call performed inside the `execute_claim` function never goes inside the if body and will never return `Err(ContractError::NoEligibilityError {})`